src/chatgpt.rs

```diff
@@ -181,7 +199,13 @@ impl GptClient {
          
         let response_object = match parse_response(&response_text) {
             Ok(response) => response,
-            Err(e) => panic!("Error while parsing response object: {}", e),
+            Err(e) => {
+                if let Ok(error_response) = parse_error_response(&response_text){
+                    panic!("Error while parsing response object: {}", error_response.error.message)
+                }else{
```

Comment:
⛏ Nitpick: Quite minor but in future it would be more "Polished" if you could run your [LSP formatter](https://github.com/divanvisagie/dotfiles/blob/main/nvim/after/plugin/lsp.lua#L92) to make sure we have a space between the keywords an the braces.